src/client/dfs/dfs.c

```diff
@@ -1075,12 +1077,14 @@ open_symlink(dfs_t *dfs, dfs_obj_t *parent, int flags, daos_oclass_id_t cid,
 		entry->value_len = value_len;
 		rc = insert_entry(parent->oh, DAOS_TX_NONE, sym->name, len,
 				  DAOS_COND_DKEY_INSERT, entry);
-		if (rc) {
+		if ((rc == EEXIST) || (rc == 0)) {
```

Comment:
we leak sym->value in this  case?

Reply:
I thought dfs_release() will free sym->value. I will double check.

Reply:
my question was on error (eexist). in that case creating the symlink has failed, and the user gets back an error.
in that case there is no dfs_release() called, and the sym object returned is NULL

Reply:
Thank you very much! I just revised dfs_sys_symlink() to avoid possible memory leak for obj. When dfs_open_stat() returns EEXIST, obj is not NULL.

Reply:
yes but we call D_FREE() there.. unfortunately i still think this is wrong

Reply:
this is a create path. so if the insert returns EEXIST, we have to call free on the value.
the mem leak i pointed out is for the symlink value not the obj.
i do not see anything wrong in the original code before your change, but again maybe im missing something.

Reply:
thx! It is fixed now.